### PR TITLE
chore(lint): enable the "unused" linter and remove code

### PR DESCRIPTION
**Pull request by janisz**
## Description

This PR enables the "unused" linter and removes code.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

CI – it's building so it should be fine unless code depend on side effects of importing package

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by porridge** [comment]
@janisz 
```
# github.com/stackrox/rox/pkg/scanners/quay [github.com/stackrox/rox/pkg/scanners/quay.test]
pkg/scanners/quay/quay_integration_test.go:73:3: undefined: log
```

**Review by porridge — APPROVED** [review]
Approving for `operator/` directory on behalf of @stackrox/draco

**Comment by porridge** [comment]
> Approving for `operator/` directory on behalf of @stackrox/draco

Actually, the linter is still failing:

```
operator/pkg/central/extensions/reconcile_scanner_db_password.go:18:2: const `scannerDBPasswordKey` is unused (unused)
	scannerDBPasswordKey = `password`
```

**Review by 0x656b694d — APPROVED** [review]

File: central/compliance/manager/run.go [hunk 1]
```diff
@@ -159,13 +154,3 @@ func (r *runInstance) ToProto() *v1.ComplianceRun {
 	}
 	return proto
 }
-
-func (r *runInstance) shouldDelete() bool {
```
Comment (RTann) [inline central/compliance/manager/run.go:163]: Should this be used?

File: central/imagecomponent/datastoretest/datastoretest.go
Comment (RTann) [inline central/imagecomponent/datastoretest/datastoretest.go:1]: what's the purpose of this?

File: central/sensor/service/connection/sensorevents.go [hunk 2]
```diff
@@ -78,10 +78,6 @@ func stripTypePrefix(s string) string {
 	return s
 }
 
-func (s *sensorEventHandler) disableReconciliation() {
```
Comment (RTann) [inline central/sensor/service/connection/sensorevents.go:81]: Curious what this was used for originally and why it's no longer needed

File: pkg/renderer/helm_values.go [hunk 3]
```diff
@@ -13,9 +13,6 @@ import (
 //go:embed templates/public_values.yaml.tpl
 var publicValuesTemplateStr string
 
-//go:embed templates/private_values.yaml.tpl
-var privateValuesYamlTemplateStr string
```
Comment (RTann) [inline pkg/renderer/helm_values.go:17]: curious why this isn't used

**Review by porridge — APPROVED** [review]
LGTM for operator/ and pkg/renderer

File: central/sensor/service/connection/sensorevents.go (hunk 2 above)
Comment (janisz) [reply central/sensor/service/connection/sensorevents.go:81]: @fredrb Could you answer this question? The only usage was removed in https://github.com/stackrox/stackrox/commit/c7c0caf59a8c02ec202a0eec2737c737176062ab#diff-5eebef2d74ae81573c60ae033445bbc96d1d4fe325735839f47c945e6f5a1c97L636

File: central/compliance/manager/run.go (hunk 1 above)
Comment (janisz) [reply central/compliance/manager/run.go:163]: It was removed after the only call was removed https://github.com/stackrox/stackrox/pull/8783/files/005bcb5619ec5f22f9369e9f2be57792535f4882#r1406507675

File: central/compliance/manager/manager_impl.go [hunk 4]
```diff
@@ -131,24 +131,6 @@ func (m *manager) createRun(domain framework.ComplianceDomain, standard *standar
 	return run
 }
 
-func (m *manager) cleanupRuns() {
```
[the remark this answers is not in the document]
Comment (janisz) [reply central/compliance/manager/manager_impl.go:134]: @connorgorman Could you answer this?
The only call was removed in https://github.com/stackrox/stackrox/commit/32c42c36d8bec066729e02f65134e3aa7e909082#diff-4ccf48e6fa9ba9b0df217e989f69b6e588da1b497e08a8af87dbfa7ee4382c03L272

File: central/sensor/service/connection/sensorevents.go (hunk 2 above)
Comment (fredrb) [reply central/sensor/service/connection/sensorevents.go:81]: @RTann there was a bug in the original client reconciliation epic that was released in 4.3. We ended up changing the approach in #8430. If you're interested in the issue you can read the doc linked in the PR.

File: central/compliance/manager/manager_impl.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (connorgorman) [reply central/compliance/manager/manager_impl.go:134]: No, at this point we never clean up compliance scans, but it was only used from the scheduling capabilities

File: central/compliance/manager/run.go (hunk 1 above)
Comment (connorgorman) [reply central/compliance/manager/run.go:163]: The comment about is about the fact that was to be used with compliance schedules that were never released

**Review by connorgorman — APPROVED** [review]

**Review by RTann — APPROVED** [review]

**Review by stehessel — APPROVED** [review]

**Review by vikin91 — APPROVED** [review]
I checked Maple-related code and it LGTM!

**Comment by janisz** [comment]
https://github.com/stackrox/stackrox/issues/7729

**Pull request merged**